packages/gestalt/src/Masonry.js

```diff
@@ -325,18 +361,110 @@ export default class Masonry<T> extends React.Component<Props<T>, State<T>> {
     this.scrollContainer = ref;
   };
 
-  fetchMore = () => {
-    const { loadItems } = this.props;
-    if (loadItems && typeof loadItems === 'function') {
-      this.setState(
-        {
-          isFetching: true,
-        },
-        () => loadItems({ from: this.props.items.length })
-      );
+  getLayout = () => {
+    const {
+      columnWidth,
+      flexible,
+      gutterWidth: gutter,
+      measurementStore,
+      minCols,
+    } = this.props;
+    const { width } = this.state;
+
+    let layout;
+    if (flexible && width !== null) {
+      layout = fullWidthLayout({
+        gutter,
+        cache: measurementStore,
+        minCols,
+        idealColumnWidth: columnWidth,
+        width,
+      });
+    } else if (
+      this.props.layout === UniformRowLayoutSymbol ||
+      this.props.layout instanceof LegacyUniformRowLayout
+    ) {
+      layout = uniformRowLayout({
+        cache: measurementStore,
+        columnWidth,
+        gutter,
+        minCols,
+        width,
+      });
+    } else {
+      layout = defaultLayout({
+        cache: measurementStore,
+        columnWidth,
+        gutter,
+        minCols,
+        width,
+      });
+    }
+
+    return layout;
+  };
+
+  calculateRenderLayout = () => {
+    const { measurementStore, minCols } = this.props;
+    const { items } = this.state;
+
+    // Full layout is possible
+    const itemsToRender = items.filter(
+      item => item && measurementStore.has(item)
+    );
+
+    const layout = this.getLayout();
+    const renderPositions = layout(itemsToRender);
+    // Math.max() === -Infinity when there are no renderPositions
+    const height = renderPositions.length
+      ? Math.max(...renderPositions.map(pos => pos.top + pos.height))
+      : 0;
+
+    const itemsToMeasure = items
+      .filter(item => item && !measurementStore.has(item))
+      .slice(0, minCols);
+    const measuringPositions = layout(itemsToMeasure);
+
+    return {
+      height,
+      itemsToRender,
+      itemsToMeasure,
+      measuringPositions,
+      renderPositions,
+    };
+  };
+
+  handleOnLayerUpdate = () => {
```

Comment:
You don't need the `on` here - just `handleLayerUpdate`.

Reply:
done